lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -101,20 +103,50 @@ class OptimizedImages extends Gatherer {
    * @return {!Promise<{originalSize: number, jpegSize: number, webpSize: number}>}
    */
   calculateImageStats(driver, networkRecord) {
-    const param = JSON.stringify(networkRecord.url);
-    const script = `(${getOptimizedNumBytes.toString()})(${param})`;
-    return driver.evaluateAsync(script).then(stats => {
-      const isBase64DataUri = networkRecord.isBase64DataUri;
-      const base64Length = networkRecord.url.length - networkRecord.url.indexOf(',') - 1;
-      return {
-        originalSize: isBase64DataUri ? base64Length : networkRecord.resourceSize,
-        jpegSize: isBase64DataUri ? stats.jpeg.base64 : stats.jpeg.binary,
-        webpSize: isBase64DataUri ? stats.webp.base64 : stats.webp.binary,
-      };
+    let uriPromise = Promise.resolve(networkRecord.url);
+
+    // Get the request body for cross-origin images to circumvent canvas CORS protections
+    if (!networkRecord.isSameOrigin && !networkRecord.isBase64DataUri) {
+      const requestId = networkRecord.requestId;
+      uriPromise = driver.sendCommand('Network.getResponseBody', {requestId}).then(resp => {
+        if (!resp.base64Encoded) {
+          throw new Error('expected image to be base64 encoded');
```

Comment:
if this ends up in the `debugString` won't this be a bit confusing for the user since the image wouldn't have been a data uri? What about just "unable to access cross-origin image body" or something since that's the important bit from their perspective.